feat: expand cryptographic algorithms into separate traits

jceb opened the pull request:
Closes #15

ottomorac commented:
Thanks @jceb this is looking good. My adjustment would be on the description from this:
>A DID method supports the cryptographic algorithm

To something like this:
>A DID method supports the cryptographic algorithm curve, which has been verified by any of the government entities mentioned in appendix X

As the idea was to still reference in the spec the list of government bodies we discussed with @msporny here as indication that the did method complies with the trait https://github.com/decentralized-identity/did-traits/issues/15#issuecomment-2543350824

jceb commented:
@ottomorac @msporny I added the list of national cryptographic standards bodies to the appendix and added a reference to the traits definition. Please review the list. I had some trouble finding the correct names and links to their websites. The two LLMs I tried (Claude and OpenAPI) came up with different lists :cry:  https://github.com/decentralized-identity/did-traits/pull/32/files#diff-c4795c497b83a8c03e33535caf0fb0e1512cecd8cb448f62467326277c152afe

ottomorac commented:
Hi @jceb good job.

Here are the missing ones so you can add them:

- Brazil:   "ITI - Instituto Nacional de Tecnologia da Informação"  - https://www.gov.br/iti/pt-br
- Russia: "EASC - Euro-Asian Council for standardization, metrology and certification" - https://easc.by/en
- South Korea:  "KCMVP - Korea Cryptographic Module Validation Program - National Intelligence Service - NIS" - https://eng.nis.go.kr/EAF/1_7_2_1.do

Sources used to confirm:
- Brasil: https://pt.wikipedia.org/wiki/Instituto_Nacional_de_Tecnologia_da_Informa%C3%A7%C3%A3o
- Russia: https://en.wikipedia.org/wiki/GOST
- South Korea: https://en.wikipedia.org/wiki/LSH_(hash_function)

jceb commented:
Awesome, thank you!
